tests/test_date.py

```diff
@@ -258,6 +259,23 @@ def test_should_parse_date(self, date_string, date_formats, expected_result):
         self.then_parsed_period_is('day')
         self.then_parsed_date_is(expected_result)
 
+    @parameterized.expand([
+        param(date_string='Fri Jan 26 16:32:21 +0000 2018', date_formats=['%a %b %d %H:%M:%S %z %Y'],
+              expected_result=datetime(2018, 1, 26, 16, 32, 21)),
+        param(date_string='15/04/1904 14:05 +0004', date_formats=['%d/%m/%Y %H:%M %z'],
+              expected_result=datetime(1904, 4, 15, 14, 5, tzinfo=timezone(timedelta(0, 240)))),
+        param(
+            date_string='20-05-1994 23:00 GMT', date_formats=['%d-%m-%Y %H:%M %Z'], expected_result=datetime(
+                1994, 5, 20, 23, 0, tzinfo=StaticTzInfo('GMT', timedelta(seconds=0))
+            )
+        ),
+    ])
+    def test_should_parse_localized_dates(self, date_string, date_formats, expected_result):
+        self.when_date_is_parsed_with_formats(date_string, date_formats)
+        self.then_date_was_parsed()
+        self.then_parsed_period_is('day')
+        self.then_parsed_date_is(expected_result)
+
```

Comment:
Are these tests complete enough?

Given the changes I would expect a test where the date contains a timezone, settings specify a different timezone, and we check that the parsed timezone is the one from the input string, and not the one from the settings.

Also a test about the change to `if settings.RETURN_AS_TIMEZONE_AWARE is not True:`.